exporter/device.go

```diff
@@ -0,0 +1,35 @@
+package exporter
+
+import (
+	"context"
+	"time"
+
+	"github.com/ffddorf/unms-exporter/client/devices"
+	"github.com/ffddorf/unms-exporter/models"
+)
+
+var defaultWithInterfaces = true
+
+func (e *Exporter) fetchDeviceData() ([]*models.DeviceStatusOverview, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	params := &devices.GetDevicesParams{
+		WithInterfaces: &defaultWithInterfaces,
+		Context:        ctx,
+	}
+	devicesResponse, err := e.api.Devices.GetDevices(params)
+	if err != nil {
+		return nil, err
+	}
+
+	data := make([]*models.DeviceStatusOverview, 0, len(devicesResponse.Payload))
+	for _, overview := range devicesResponse.Payload {
+		if overview.Identification == nil {
+			continue
+		}
```

Comment:
this check now seems to happen twice? i still see it in the main exporter logic?

Reply:
Right, the one in `collectImpl()` is redundant.